src/txn/txn_recover.c

```diff
@@ -777,11 +777,13 @@ __recovery_file_scan_prefix(WT_RECOVERY *r, const char *prefix, const char *igno
 {
     WT_CURSOR *c;
     WT_DECL_RET;
+    uint64_t file_count;
     int cmp;
     const char *uri, *config;
 
     /* Scan through all entries in the metadata matching the prefix. */
     c = r->files[0].c;
+    file_count = 0;
```

Comment:
Instead of counting files here, have you considered counting them in `__wt_rts_btree_apply_all`? You can walk the metadata twice, once to gather the total count and the second time to actually do the work. In this way, you would not need to add a new variable to `WT_CONNECTION` and all the relevant logic would be in one place.

No need to change the code, but I'm curious why you chose this approach. (And depending on the answer, whether it would justify adding a comment somewhere.)

Thank you!

Reply:
I chose this way specifically to avoid walking the metadata another time. I agree with the code locality. Most of the places where this is an issue have thousands and thousands of tables and we're already walking the metadata in recovery and have exactly the information we need at that time. Maybe in the scheme of the work RTS has to do, walking the entire metadata another time should not be a consideration. 

I can code up that approach and see what it looks like. Perhaps its simplicity will outweigh another metadata walk.

I'm happy to add a comment somewhere. Please suggest what you'd find helpful.